resources/service_systemd.rb

```diff
@@ -0,0 +1,82 @@
+provides :nexus3_service_systemd
+
+provides :nexus3_service, os: 'linux' do |_node|
+  Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+end
+
+property :instance_name, String, name_property: true
+property :install_dir, String
+property :nexus3_user, String
+property :nexus3_group, String
+
+action :start do
+  create_init
+
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports restart: true, status: true
+    action :start
+    only_if 'command -v java >/dev/null 2>&1 || exit 1'
+  end
+end
+
+action :stop do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :stop
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action :restart do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :restart
+  end
+end
+
+action :disable do
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :disable
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action :enable do
+  create_init
+
+  service "nexus3_#{new_resource.instance_name}" do
+    provider Chef::Provider::Service::Systemd
+    supports status: true
+    action :enable
+    only_if { ::File.exist?("/etc/systemd/system/nexus3_#{new_resource.instance_name}.service") }
+  end
+end
+
+action_class do
+  def create_init
+    template "/etc/systemd/system/nexus3_#{new_resource.instance_name}.service" do
```

Comment:
Ok Systemd is starting to get really annoying. It will not accept a variable as its command for ExecStart / ExecStop. Using ExecStart=/bin/sh -c $START_COMMAND instead + a User=Nexus fails to start the service. I have not found a way that starts it properly apart from having ExecStart=/opt/nexus-3.3.1-01/bin/nexus start in the service file, which renders this whole thing useless.

I suggest to make it simple and create a unit file for every instance of Nexus we want to bring up. I suspect it will not really be a big issue (if we find someone willing to bring up more than 3 different instances of Nexus on the same machine, we can offer them all our sympathies and suggest changing careers).